### PR TITLE
Cluster stability

### PR DESCRIPTION
**Pull request by marcinwadon**
Tested on the cluster using 3 nodes. Cluster were stable and aligned but after some time (~2h) transactions necessary for consensus/snapshot somehow vanished (cache eviction?).
Let's merge it and make it our baseline to debug next issues on top of that.

### CODE REVIEW

File: src/main/scala/org/constellation/p2p/DataResolver.scala [hunk 1]
```diff
@@ -170,16 +174,16 @@ class DataResolver {
...
 
 }
 
 object DataResolver extends DataResolver
 
-case class DataResolutionOutOfPeers(endpoint: String,hash: String, peers: Iterable[String])
+case class DataResolutionOutOfPeers(thisNode: String, endpoint: String,hash: String, peers: Iterable[String])
     extends Exception(
-      s"Run out of peers when resolving: $endpoint with hash: $hash following tried: $peers"
+      s"node [${thisNode}] Run out of peers when resolving: $endpoint with hash: $hash following tried: $peers"
```
Comment (Nikolaj-K) [inline src/main/scala/org/constellation/p2p/DataResolver.scala]: What's the value of the `endpoint` val typtically? I just found `val endpoint: String = "endpoint"`. Extrapolating on the style, I'd expect `resolving: endpoint $endpoint` here. 

Sorry for the superficial notes, I'll get deeper when I've an endpoint to quiery.

Comment (marcinwadon) [reply src/main/scala/org/constellation/p2p/DataResolver.scala]: This change is only about adding debugging info. We're kinda unable to think about and fix all places we're touching :)

File: src/main/scala/org/constellation/consensus/Round.scala [hunk 2]
```diff
@@ -295,13 +322,22 @@ class Round(roundData: RoundData, arbitraryTransactions: Seq[(Transaction, Int)]
...
 
+      dao.metrics.incrementMetric("acceptMajorityCheckpointBlockSelectedCount_" + selectedCheckpointBlocks.size)
+
       val finalFacilitators = selectedCheckpointBlocks.keySet.map(_.id).toSet
-      val cache = CheckpointCache(Some(checkpointBlock),
-        height = checkpointBlock.calculateHeight())
-      broadcastSignedBlockToNonFacilitators(FinishedCheckpoint(cache, finalFacilitators))
+      val cache = CheckpointCache(Some(checkpointBlock), height = checkpointBlock.calculateHeight())
+      log.debug(s"[${dao.id.short}] accepting majority checkpoint block ${checkpointBlock.baseHash}  " +
+        s" with txs ${checkpointBlock.transactions.map(_.hash)} " +
```
Comment (Nikolaj-K) [inline src/main/scala/org/constellation/consensus/Round.scala]: `baseHash} _ _ " + s" _ with `
Not relevant but this looks like it's an artifact of a previous edit.

File: src/main/scala/org/constellation/p2p/DataResolver.scala (hunk 1 above)
Comment (Nikolaj-K) [reply src/main/scala/org/constellation/p2p/DataResolver.scala]: I know the feel

File: src/main/scala/org/constellation/consensus/Round.scala (hunk 2 above)
Comment (marcinwadon) [reply src/main/scala/org/constellation/consensus/Round.scala]: nope, it's fine. https://docs.scala-lang.org/overviews/core/string-interpolation.html

Comment (marcinwadon) [reply src/main/scala/org/constellation/consensus/Round.scala]: or you mean - spaces? it's `log.debug` so it's not so important

File: src/main/scala/org/constellation/p2p/DataResolver.scala (hunk 1 above)
Comment (wkoszycki) [reply src/main/scala/org/constellation/p2p/DataResolver.scala]: endpoint will be either `checkpoint` or `transaction`

**Pull request merged**